src/TestSuite/Schema/SchemaCleaner.php

```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Schema;
+
+use Cake\Console\ConsoleIo;
+use Cake\Database\Schema\BaseSchema;
+use Cake\Database\Schema\CollectionInterface;
+use Cake\Datasource\ConnectionInterface;
+use Cake\Datasource\ConnectionManager;
+use Cake\TestSuite\TestConnectionManager;
+
+/**
+ * This class will help dropping all tables of a given connection
+ * and truncate the non phinx migrations tables.
+ */
+class SchemaCleaner
+{
+    /**
+     * Drop all tables of the provided connection.
+     *
+     * @param string $connectionName Name of the connection.
+     * @param \Cake\Console\ConsoleIo|null $io Console IO to output the processes.
+     * @return void
+     * @throws \Exception if the dropping failed.
+     */
+    public static function drop(string $connectionName, ?ConsoleIo $io = null)
```

Comment:
Any reason these methods need to be static? We could make an instance and pass in the ConsoleIo as a constructor parameter. It makes the other methods a bit simpler.

Reply:
These methods could be called within setUp or tearDown of tests easily if they are static.
If a given test modifies intentionally the schema, the developer can decide to rebuild the schema by dropping it and re-running the Schema construction (with SchemaManager or Mogrator).

This is the motivation for these being static. The $io is there more for the SchemaManager to pass on its own $io.

Does this make sense?

Reply:
ps: I am also O.K. to have the schema cleaner methods non-static, especially if you think that it would more fit the Cake spirit.

Reply:
My personal bias is to avoid static methods unless they are the only way of doing something. These methods could be exposed via the `FixtureManager` if we require a more convenient way to access them. I suspect that dropping schema is a rare task in applications and it doesn't need to be 'easy'.